src/coreclr/jit/optcse.cpp

```diff
@@ -2601,10 +2601,17 @@ class CSE_Heuristic
         //
         if (candidate->LiveAcrossCall())
         {
+            if (candidate->Expr()->IsCnsFltOrDbl() && (CNT_CALLEE_SAVED_FLOAT == 0))
+            {
+                // We should do CSE for fp constants in case of LiveAcrossCall only when absolutely necessary
+                // on ABIs without callee-saved registers.
+                cse_use_cost += 2;
+            }
```

Comment:
I am ok with this more surgical fix for 6.0, but if we decide to go this route then you should open an issue about revisiting the float heuristics for .NET 7. It does not really make sense to adjust the use cost of a CSE based on the contents of the tree that is being CSE'd, I think.